telemetrylib/TCP.cpp

```diff
@@ -62,14 +66,38 @@ public slots:
         {
             QTcpSocket* sender = static_cast<QTcpSocket*>(QObject::sender());
             _sockets.removeOne(sender);
-            connected = false;
+            connection = false;
             emit connectionStatusChanged();
         }
     }
```

Comment:
Can we add something like this to remove all sockets when we lose connection in your check below? We'll only ever have one, so we can just remove them all